src/pybind/mgr/orchestrator/module.py

```diff
@@ -1544,3 +1554,14 @@ def _upgrade_stop(self):
         self._orchestrator_wait([completion])
         raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
+
+    @_cli_write_command(
+        'orch ok-to-stop',
+        "name=daemon_type,type=CephString "
+        "name=daemon_ids,type=CephString,n=N",
+        desc='Check if the specified daemons can be safely stopped without reducing availability')
+    def _ok_to_stop(self, daemon_type: str, daemon_ids: List[str]):
+        completion = self.ok_to_stop(daemon_type, daemon_ids)
+        self._orchestrator_wait([completion])
+        raise_if_exception(completion)
+        return HandleCommandResult(stdout=completion.result_str())
```

Comment:
It still feels odd. I mean, looking at https://docs.ceph.com/docs/master/api/mon_command_api/ we have:

* [`ceph mds ok-to-stop`](https://docs.ceph.com/docs/master/api/mon_command_api/#mds-ok-to-stop)
* [`ceph mon ok-to-stop`](https://docs.ceph.com/docs/master/api/mon_command_api/#mon-ok-to-stop)
* [`ceph osd ok-to-top`](https://docs.ceph.com/docs/master/api/mon_command_api/#osd-ok-to-stop)

Woudn't it make more sense to provide commands in a similar fashion? Like

* `ceph nfs ok-to-stop` 
* `ceph rgw ok-to-stop` 

etc?

Reply:
yeah that's redundant. if we want to add orchestrator specific logic we should either extend the the existing mds/mon/osd commands or add the nfs/rgw ones etc.